pkg/config/config.go

```diff
@@ -369,8 +361,8 @@ func (m Module) listUnusedModules() []ModuleID {
 	used := map[ModuleID]bool{}
 	// Recurse through objects/maps/lists checking each element for having `ProductOfModuleUse` mark.
 	cty.Walk(m.Settings.AsObject(), func(p cty.Path, v cty.Value) (bool, error) {
-		if mark, has := HasMark[ProductOfModuleUse](v); has {
-			used[mark.Module] = true
+		for _, mod := range IsProductOfModuleUse(v) {
```

Comment:
The way I read the code you are migrating from a list of Marks to a single Mark that contains a string that is parsable into a list?

Reply:
Previously there was at most one module per value, now it's a list of modules.
I've opted to a parsable list (hidden implementation detail) because the `Mark` struct has to be `hashable` and `slices` aren't hashable.